server/middlewares/token.js

```diff
@@ -0,0 +1,46 @@
+// modules
+const jwt = require('jsonwebtoken');
+
+exports.validate = (req, res, next) => {
+  const token = req.headers['authorization'];
+
+  if (token) {
+    // grab token from headers request
+    const headersToken = token.split(" ")[1];
+
+    jwt.verify(headersToken, "secretKey", err => {
```

Comment:
I was thinking about `secretKey` yesterday, using user's password.

You'd have to query users' password every time to validate token. That'd basically make every query expensive.

Also as people on Hashnode mentioned, users' passwords aren't secure anyway.

If you want to keep secretekey different for everyone, you could generate a random key and store that in DB. And use that to sign the token and to verify. This can be different for everyone.

Reply:
> If you want to keep `secretekey` different for everyone, you could generate a random key and store that in DB. And use that to sign the token and to verify. This can be different for everyone.

How about using email as secret key? Since each account as a unique email address.

Reply:
What? No, email ID is public data. Anybody who knows your email ID, has you token forever, that's what it means.

Reply:
Hmm... 🤔

Reply:
How about.. We get the timestamp at the time of member login and store inside `member` table _(maybe that data can be later helpful for creating analytics dashboard for admins, anyway)_.

Then, at the time of token validation we can get the `login_request_timestamp` _(name of the database column)_ and decode the token.

How this sounds? It follows what you mentioned about generating a random key and storing inside the database.

Reply:
But timestamp isn't random though. Since this is an open source project, anybody can look at the code and see that it's using timestamp, Since we are expiring token every 2days, user has to login again every 2 days, what that means is, someone can generate token for every second for 2 day period and try using them all as part of brute force method. This isn't hard to do programmatically, and it would hardly take an hour or even less. And if that someone can guess the login time roughly, say maybe he/she knows someone logged in could of hour ago, brute force method would yield results even quicker.

Reply:
So, we should use UUID to generate a `secretKey`?